deepchem/models/tests/test_jax_model.py

```diff
@@ -87,12 +89,191 @@ def bce_loss(pred, tar, w):
 
   # JaxModel Working
   j_m = JaxModel(
-      model,
+      model.apply,
       params,
       criterion,
       output_types=['loss', 'prediction'],
       batch_size=256,
       learning_rate=0.001,
       log_frequency=2)
-  results = j_m.fit(dataset, nb_epochs=50, deterministic=True)
-  assert results < 1.0
+  _ = j_m.fit(dataset, nb_epochs=25, deterministic=True)
+  scores = j_m.evaluate(dataset, [metric])
+  assert scores[metric.name] > 0.8
+
+
+@pytest.mark.jax
+def test_overfit_subclass_model():
+  """Test fitting a JaxModel defined by subclassing Module."""
+  n_data_points = 10
+  n_features = 2
+  np.random.seed(1234)
+  X = np.random.rand(n_data_points, n_features)
+  y = (X[:, 0] > X[:, 1]).astype(np.float32)
+  dataset = dc.data.NumpyDataset(X, np.expand_dims(y, axis=1))
+
+  class Encoder(hk.Module):
+
+    def __init__(self, output_size: int = 1):
+      super().__init__()
+      self._network = hk.nets.MLP([512, 256, 128, output_size])
+
+    def __call__(self, x: jnp.ndarray):
+      x = self._network(x)
+      return x, jax.nn.sigmoid(x)
+
+  def f(x):
+    net = Encoder(1)
+    return net(x)
+
+  # Model Initilisation
+  model = hk.transform(f)
```

Comment:
If every `JaxModel` requires the model to be wrapped as a function, does it make sense to allow the user to input an `hk.Module` (which will be familiar to torch and tensorflow users) and then automatically wrap it as a function and do the transform into a model?

Reply:
Yes, I can add a utility function that does it internally.